src/newsfeed/infrastructure/utils.py

```diff
@@ -0,0 +1,13 @@
+"""Utils module for infrastructure."""
+
+from urllib.parse import urlparse, parse_qsl
+
+
+def parse_redis_dsn(dsn):
+    """Redis dsn parser."""
+    parsed_dsn = urlparse(dsn)
+    assert parsed_dsn.scheme in ('', 'redis'), ('Unsupported URI scheme', parsed_dsn.scheme)
```

Comment:
Can you describe the case when `scheme == ''`?

Reply:
I don't think so :) Fixed.